pkg/assembler/nodes.go

```diff
@@ -16,9 +16,161 @@
 package assembler
 
 // ArtifactNode is a node that represents an artifact
-// TODO: implement GuacNode
-type ArtifactNode struct{}
+type ArtifactNode struct {
+	Name     string
+	Checksum string
+}
+
+func (an ArtifactNode) Type() string {
+	return "Artifact"
```

Comment:
nit: i have a slight preference for camel case or all lowercase-with-hypens.

Reply:
Neo4j uses all capitals for node/edge types, but that would be too much. Let's leave like this for now as it will probably get replaced with the serialization via field tags later.